app/waf/src/services/asg.service.ts

```diff
@@ -312,26 +336,58 @@ export class ASGManager {
     );
   }
 
-  async icontrolGet(trustDeviceId: string, path: string): Promise<object> {
+  async installCert(trustDeviceId: string, body: object): Promise<object> {
+    return await this.icontrolPost(
+      trustDeviceId,
+      '/mgmt/tm/sys/crypto/cert',
+      body,
+    );
+  }
+  async installKey(trustDeviceId: string, body: object): Promise<object> {
+    return await this.icontrolPost(
+      trustDeviceId,
+      '/mgmt/tm/sys/crypto/key',
+      body,
+    );
+  }
+
+  async uploadFile(
+    trustDeviceId: string,
+    content: string,
+    length: number,
+    name: string,
+  ): Promise<object> {
+    return await this.service.uploadFile(
```

Comment:
Maybe we can consider to reuse icontrolPost(), instead of creating a new function, but that depends on you.

Reply:
I removed these new functions to keep the logic clear.

This is due to we developed in parallel.